temporal/tracing.go

```diff
@@ -1,11 +1,136 @@
 package temporal
 
 import (
+	"context"
+	"github.com/samber/lo"
 	temporalotel "go.temporal.io/sdk/contrib/opentelemetry"
+	"go.temporal.io/sdk/converter"
 	"go.temporal.io/sdk/interceptor"
+	"go.temporal.io/sdk/workflow"
+	"strconv"
 )
 
+type (
+	workflowTraceParametersKey        struct{}
+	workflowTraceParametersPropagator struct{}
+
+	TaskResult[T any] struct {
+		Result   T
+		Err      error
+		Attempts int
+		Status   string
+	}
+)
+
+const tracingKey = "armory-tracing"
+
+func (w workflowTraceParametersPropagator) Inject(ctx context.Context, writer workflow.HeaderWriter) error {
+	kvp, ok := ctx.Value(workflowTraceParametersKey{}).(*map[string]string)
```

Comment:
Can you help me understand why the value needs to be a pointer to a map? A Go map is already mutable.

Reply:
I would avoid a pointer to a map unless there's a really compelling justification because it adds cognitive overhead

Reply:
it doesn't. that's my bad - i forgot that map or array is already a pointer.